graph/vertex/impl/ThingVertexImpl.java

```diff
@@ -98,11 +98,17 @@ public boolean isWrite() {
         throw TypeDBException.of(INVALID_THING_VERTEX_CASTING, className(ThingVertex.Write.class));
     }
 
+    @Override
+    public int compareTo(ThingVertex o) {
+        return iid.bytes().compareTo(o.iid().bytes());
+    }
+
```

Comment:
thing vertices, which we want to sort, and directed edges are all sortable and therefore require `compareTo`